### PR TITLE
Fix for Windows Environments

### PR DESCRIPTION
**Pull request by AdrianoCahete**
Fixes colors and marks for Windows OS #1228 

colors

marks

For tree visualization, code is still here, but i don't know why isn't working well. I think it's a good fix, but not the better. Feel free to help in that point.

### CODE REVIEW

**Comment by XhmikosR** [comment]
I had opened https://github.com/GoogleChrome/lighthouse/pull/1279 which isn't as complete as this one so I'm gonna close that.

But there's also https://github.com/GoogleChrome/lighthouse/pull/1272 before your PR which fixes debug issues.

File: lighthouse-cli/printer.ts [hunk 1]
```diff
@@ -56,9 +57,15 @@ function formatAggregationResultItem(score: boolean | number | string, suffix?:
   const reset = '\x1B[0m';
 
   if (typeof score === 'boolean') {
-    const check = `${green}✓${reset}`;
-    const fail = `${red}✘${reset}`;
-    return score ? check : fail;
+    if (os.platform() == 'win32') {
```
Comment (XhmikosR) [inline lighthouse-cli/printer.ts]: This returns the same so the return call should be out of the if check.

**Comment by XhmikosR** [comment]
Also, you should make sure lint is passing, which doesn't at the moment.

File: lighthouse-core/formatters/critical-request-chains.js [hunk 2]
```diff
@@ -159,9 +164,16 @@ class CriticalRequestChains extends Formatter {
 
         // Create the appropriate tree marker based on the depth of this
         // node as well as whether or not it has children and is itself the last child.
-        const treeMarker = ancestorTreeMarker +
+        const treeMarker = ancestorTreeMarker;
+        if (os.platform() == 'win32') {
+          const treeMarker = ancestorTreeMarker +
```
Comment (XhmikosR) [inline lighthouse-core/formatters/critical-request-chains.js]: Change this to `treeMarker += foo`.

File: lighthouse-core/formatters/critical-request-chains.js [hunk 3]
```diff
@@ -159,9 +164,16 @@ class CriticalRequestChains extends Formatter {
...
+            (isLastChild ? '\u2514\u2500' : '\u251C\u2500') +
+            (hasChildren ? '\u252C' : '\u2500');
+        } else {
+          const treeMarker = ancestorTreeMarker +
```
Comment (XhmikosR) [inline lighthouse-core/formatters/critical-request-chains.js]: Same here.

**Comment by XhmikosR** [comment]
If the maintainers agree, I could make a proper PR by cherry picking #1272 and this one, fixing lint issues and keep author info which IMO is the most important part this PR is missing.

Then we could close those 2 PRs at once.

**Comment by XhmikosR** [comment]
Made #1307 which has this PR fixed but kept author information too.

**Comment by AdrianoCahete** [comment]
> Made #1307 which has this PR fixed but kept author information too.

👍

**Pull request closed without merging**